### PR TITLE
Cockpit 168

### PR DESCRIPTION
**Pull request by martinpitt**
Rendered: https://martinpitt.github.io/cockpit-project.github.io/blog/cockpit-168.html

### CODE REVIEW

**Comment by garrett** [comment]
Modifications in a PR @ https://github.com/martinpitt/cockpit-project.github.io/pull/2

File: _posts/2018-05-16-cockpit-168.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+Cockpit is the [modern Linux admin interface](https://cockpit-project.org/). We release regularly.
+Here are the release notes from version 168.
+
+### Improve checks for root privilege availability
+
+Many actions, like rebooting the machine or joining an IPA domain, can only be
+done by an administrator (`root`). To decide whether to enable these actions,
+Cockpit previously checked if the user was a member of the `wheel`
+(RHEL/Fedora) or `sudo` (Debian) groups.
```
Comment (garrett) [inline _posts/2018-05-16-cockpit-168.md]: Many actions, like rebooting or joining an IPA domain, can only be performed by
an administrator (aka: `root`).

*[IPA]: identity management system ("Identity, Policy, Audit")
*[aka]: also known as

File: _posts/2018-05-16-cockpit-168.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+This was insufficient: Other operating systems or local customizations might
+use other groups, or have user instead of group based sudo rules. For example
+with FreeIPA central user management, sudo access is given through the
+`admins@domain.name` group.
```
Comment (garrett) [inline _posts/2018-05-16-cockpit-168.md]: Cockpit previously checked if the user was a member of the `wheel` (used in
Red Hat Enterprise Linux, CentOS, and Fedora) or `sudo` (used in Debian and
Ubuntu) groups to enable these actions. Simple group checking was insufficient,
as different group names are used by other operating systems and configurations
or a system might be set up to rely on custom user-based sudo rules.

File: _posts/2018-05-16-cockpit-168.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+Cockpit now does not make any assumptions about special groups any more, and
+instead just checks if the user is capable of running commands as `root`. Most
+notably this makes privileged operations work out of the box with FreeIPA.
```
Comment (garrett) [inline _posts/2018-05-16-cockpit-168.md]: Cockpit no longer makes assumptions about special groups. Now, Cockpit simply
checks if the user is capable of running commands as `root`.

As a result of this fix, privileged operations now work by default with
[FreeIPA](https://freeipa.org/), as it uses an `admins@domain.name` group.

File: _posts/2018-05-16-cockpit-168.md (hunk 1 above)
Comment (martinpitt) [reply _posts/2018-05-16-cockpit-168.md]: This sounds a bit ambiguous, as one does not "reboot an IPA domain". One could flip it around: "Many actions, like joining an IPA domain or rebooting, can only..."

**Comment by martinpitt** [comment]
Thanks @garrett! I applied these with some minor adjustments, like the "reboot and join" issue, adding a missing comma for better legibility, and some other tiny changes.

**Comment by martinpitt** [comment]
Release is done, merging.

**Pull request merged**